src/index.js

```diff
@@ -1,45 +1,89 @@
-    const icon = document.createElement('div');
-    icon.innerHTML = `
-  <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="w-6 h-6">
-    <path stroke-linecap="round" stroke-linejoin="round" d="M12 6.75a.75.75 0 110-1.5.75.75 0 010 1.5zM12 12.75a.75.75 0 110-1.5.75.75 0 010 1.5zM12 18.75a.75.75 0 110-1.5.75.75 0 010 1.5z" />
-  </svg>
-  `;
-
-    description.appendChild(checkBox);
-    description.appendChild(descriptionText);
-    todoElement.appendChild(description);
-    todoElement.appendChild(icon);
-    todoElement.classList.add('component');
-    todoListTag.appendChild(todoElement);
-  });
+import {
+  tasks, addTask, deleteTask, editTask, toggleTaskCompletion,
+} from './todoFunctions.js';
+
+const renderTasks = () => {
+  const app = document.getElementById('app');
+  app.innerHTML = ''; // Clear the previous content
+
+  const list = document.createElement('ul');
+  list.classList.add('todo-list');
+
+  tasks
+    .sort((a, b) => a.index - b.index)
+    .forEach((task, index) => {
+      const listItem = document.createElement('li');
+      listItem.classList.add('component');
+      listItem.classList.add('todo-item');
+
+      const description = document.createElement('div');
+      description.classList.add('todo-item-description');
+
+      // Create a checkbox to toggle completion status
+      const checkbox = document.createElement('input');
+      checkbox.type = 'checkbox';
+      checkbox.checked = task.completed;
+      checkbox.addEventListener('change', () => {
+        toggleTaskCompletion(index);
+        renderTasks(); // Update the list after toggling completion status
+      });
+
+      // Create a text input for editing the task description
+
+      const input = document.createElement('input');
+      input.type = 'text';
+      input.classList.add('todo-description-text');
+      input.value = task.description;
+      input.addEventListener('input', (e) => {
+        editTask(index, e.target.value);
+      });
+
+      description.appendChild(checkbox);
+      description.appendChild(input);
+
+      // Add an event listener for the Enter key press to finish editing
+      input.addEventListener('keyup', (e) => {
+        if (e.key === 'Enter') {
+          input.blur(); // Remove focus from the input to finish editing
+        }
+      });
+
+      // Create a delete button for removing the task
+      const deleteButton = document.createElement('button');
+      deleteButton.classList.add('remove-btn');
+      deleteButton.innerHTML = `
+        <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor" class="w-6 h-6">
+          <path stroke-linecap="round" stroke-linejoin="round" d="M12 6.75a.75.75 0 110-1.5.75.75 0 010 1.5zM12 12.75a.75.75 0 110-1.5.75.75 0 010 1.5zM12 18.75a.75.75 0 110-1.5.75.75 0 010 1.5z" />
+        </svg>
+      `;
```

Comment:
- [ Optional ] I think it would be better if your application worked exactly as illustrated in the project video, I noticed that when I click on the vertical `ellipsis` it does not change to a `delete` icon. The `ellipsis` icon is not associated with the `delete` action and so it would be wise that when a user clicks it, the delete icon is displayed to prompt a user to delete a todo from the list. Please see the screenshot below for more details on this.

## Project wireframe
![Screenshot from 2023-07-22 02-04-04](https://github.com/thasipablo/todo_list/assets/16392046/56295eeb-3a84-4925-b18a-9145ba5f3bdc)


## Your design
![Screenshot from 2023-07-22 01-54-57](https://github.com/thasipablo/todo_list/assets/16392046/6afd8e5d-b799-407c-8c3b-7c3a5d4ab441)